### PR TITLE
Fix parsing of shell.homepage

### LINKED ISSUE #17290: Windows nightly fails on open
Running the current Windows nightly, whether .msi or .zip install, fails to run.  The servo screen closes immediately on open.

Nothing is printed to the terminal when this occurs.

### PR DESCRIPTION
**Pull request by sadmansk**
This should fix the nightly builds crashing, although I don't know how to verify whether my fixes worked since I don't know how to test packaging. Any pointers on how to do that would be highly appreciated. 

@paulrouget Please let me know if there are any problems with these changes. 

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #17290

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17312)

### CODE REVIEW

File: ports/servo/main.rs [hunk 1]
```diff
@@ -147,9 +147,13 @@ fn main() {
 
     // If the url is not provided, we fallback to the homepage in PREFS,
     // or a blank page in case the homepage is not set either.
-    let target_url = opts::get().url.clone()
-                .unwrap_or(ServoUrl::parse(PREFS.get("shell.homepage").as_string()
-                    .unwrap_or("about:blank")).unwrap());
+    let cwd = env::current_dir().unwrap();
+    let cmdline_url = opts::get().url.clone();
+    let pref_url = PREFS.get("shell.homepage").as_string()
+        .map(|str| parse_url_or_filename(&cwd, str)).unwrap();
```
Comment (paulrouget) [inline ports/servo/main.rs]: Do not unwrap now, it's going to crash if it fails.

File: ports/servo/main.rs [hunk 2]
```diff
@@ -147,9 +147,13 @@ fn main() {
...
+    let blank_url = ServoUrl::parse("about:blank").unwrap();
+
+    let target_url = cmdline_url.unwrap_or(pref_url.unwrap_or(blank_url));
```
Comment (paulrouget) [inline ports/servo/main.rs]: Do no unwrap pref_url and blank_url, and use:

```rust
let target_url = cmdline_url.or(pref_url).or(blank_url).unwrap()
```

**Review by paulrouget — COMMENTED** [review]
`./mach package` to create the package.

**Comment by paulrouget** [comment]
Actually, to test, you can just do `./mach run -b`. This should use the pref.

**Comment by sadmansk** [comment]
@paulrouget so `pref_url_opt` holds a `Option<Option<ServoUrl>>` value, should I unwrap it once during definition or leave it as it is right now?

**Comment by paulrouget** [comment]
To avoid the Option<Option<…>>, use `and_then()`:

```rust
     let cwd = env::current_dir().unwrap();
     let cmdline_url = opts::get().url.clone();
     let pref_url = PREFS.get("shell.homepage").as_string()
         .and_then(|str| parse_url_or_filename(&cwd, str).ok());
     let blank_url = ServoUrl::parse("about:blank").ok();
     let target_url = cmdline_url.or(pref_url).or(blank_url).unwrap();
```

**Comment by sadmansk** [comment]
@paulrouget made those changes, squashed and rebased. Let me know if there is anything else.

**Comment by paulrouget** [comment]
This is no need to add `_opt` at the end of the name of your variable. We do that in some other part of the code, but here I don't think it's necessary.

**Comment by sadmansk** [comment]
@paulrouget Renamed the variables back, changes are ready now.

**Pull request merged**